daft/runners/tqdm.py

```diff
@@ -0,0 +1,10 @@
+from __future__ import annotations
+
+
+def is_running_from_ipython():
+    try:
+        from IPython import get_ipython
+
+        return get_ipython() is not None
+    except Exception:
```

Comment:
Should we be more specific?

```suggestion
    except ImportError:
```